src/telemetry/Manager.cc

```diff
@@ -38,6 +43,232 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+	static auto rt = zeek::id::find_type<zeek::RecordType>("Telemetry::MetricOpts");
+	auto r = make_intrusive<zeek::RecordVal>(rt);
+
+	static auto rt_off_prefix = rt->FieldOffset("prefix");
+	static auto rt_off_name = rt->FieldOffset("name");
+	static auto rt_off_helptext = rt->FieldOffset("helptext");
+	static auto rt_off_unit = rt->FieldOffset("unit");
+	static auto rt_off_is_total = rt->FieldOffset("is_total");
+	static auto rt_off_labels = rt->FieldOffset("labels");
+	static auto rt_off_bounds = rt->FieldOffset("bounds");
+	static auto rt_off_metric_type = rt->FieldOffset("metric_type");
+
+	r->Assign(rt_off_prefix, make_intrusive<zeek::StringVal>(prefix()));
+	r->Assign(rt_off_name, make_intrusive<zeek::StringVal>(name()));
+	r->Assign(rt_off_helptext, make_intrusive<zeek::StringVal>(helptext()));
+	r->Assign(rt_off_unit, make_intrusive<zeek::StringVal>(unit()));
+	r->Assign(rt_off_is_total, val_mgr->Bool(is_sum()));
+
+	auto label_names_vec = make_intrusive<zeek::VectorVal>(svt);
+	int i = 0;
+	for ( auto const& l : label_names() )
+		label_names_vec->Assign(i++, make_intrusive<StringVal>(l));
+	r->Assign(rt_off_labels, label_names_vec);
+
+	auto et = zeek::id::find_type<zeek::EnumType>("Telemetry::MetricType");
+	zeek_int_t metric_type_int = -1;
+
+	// This is mapping internal enum values for the types to the
+	// right EnumVal. TODO: There might be a better way.
+	switch ( metric_type_ )
+		{
+		case DOUBLE_COUNTER:
+			metric_type_int = et->Lookup("Telemetry::DOUBLE_COUNTER");
+			break;
+		case INT_COUNTER:
+			metric_type_int = et->Lookup("Telemetry::INT_COUNTER");
+			break;
+		case DOUBLE_GAUGE:
+			metric_type_int = et->Lookup("Telemetry::DOUBLE_GAUGE");
+			break;
+		case INT_GAUGE:
+			metric_type_int = et->Lookup("Telemetry::INT_GAUGE");
+			break;
+		case DOUBLE_HISTOGRAM:
+			metric_type_int = et->Lookup("Telemetry::DOUBLE_HISTOGRAM");
+			break;
+		case INT_HISTOGRAM:
+			metric_type_int = et->Lookup("Telemetry::INT_HISTOGRAM");
+			break;
+		}
+
+	if ( metric_type_int < 0 )
+		reporter->FatalError("Unable to lookup %d", int(metric_type_));
+
+	zeek::EnumValPtr metric_type_val = et->GetEnumVal(metric_type_int);
+	r->Assign(rt_off_metric_type, metric_type_val);
+
+	// TODO: Handle bounds for histograms
+	// if (metric_type_ == DOUBLE_HISTOGRAM || metric_type == INT_HISTOGRAM)
+
+	return r;
+	}
+
+zeek::RecordValPtr Manager::Metric::get_metric_record()
+	{
+	static auto svt = zeek::id::find_type<zeek::VectorType>("string_vec");
+	static auto rt = zeek::id::find_type<zeek::RecordType>("Telemetry::Metric");
+	static auto rt_off_opts = rt->FieldOffset("opts");
+	static auto rt_off_label_values = rt->FieldOffset("label_values");
+	static auto rt_off_value = rt->FieldOffset("value");
+	static auto rt_off_count_value = rt->FieldOffset("count_value");
+
+	auto r = make_intrusive<zeek::RecordVal>(rt);
+	// TODO: Cache this somewhere. Maybe do not set it here at all.
```

Comment:
Seems the manager can maintain a cache of the opts records. Either create them here on demand on 1st use, or pre-compute elsewhere.